test/e2e/test_tools.py

```diff
@@ -191,123 +194,161 @@ def test_get_project(self) -> None:
+            check=True,
+            shell=True,
+            capture_output=True,
+            cwd=self.cwd,
         )
-        self.assertTrue(os.path.isfile(id2iri_replaced_xml_filename))
 
-        result_replaced = xml_upload(
-            input_file=id2iri_replaced_xml_filename,
-            server=self.server,
-            user=self.user,
-            password=self.password,
-            imgdir=self.imgdir,
-            sipi=self.sipi,
-            verbose=True,
-            incremental=True,
-            save_metrics=False,
-            preprocessing_done=False,
+        mapping_file = list(Path(self.cwd).glob("test-data-systematic_id2iri_mapping_*.json"))[0]
+        xml_file_orig = Path("testdata/id2iri/test-id2iri-data.xml")
+        xml_file_replaced = Path("testdata/tmp/_test-id2iri-replaced.xml")
+        cmd_base = "poetry run dsp-tools id2iri --verbose"
+        subprocess.run(
+            f"{cmd_base} --outfile {xml_file_replaced.absolute()} {xml_file_orig.absolute()} {mapping_file.absolute()}",
+            check=True,
+            shell=True,
+            capture_output=True,
+            cwd=self.cwd,
         )
-        self.assertTrue(result_replaced)
-        self.assertTrue(all(not f.name.startswith("stashed_text_properties_") for f in os.scandir(".")))
-        self.assertTrue(all(not f.name.startswith("stashed_resptr_properties_") for f in os.scandir(".")))
 
-        os.remove(mapping_file)
-        os.remove(id2iri_replaced_xml_filename)
+        subprocess.run(
+            f"poetry run dsp-tools xmlupload --incremental -v {xml_file_replaced.absolute()}",
+            check=True,
+            shell=True,
+            capture_output=True,
+            cwd=self.cwd,
+        )
+        self.assertListEqual(list(Path(self.cwd).glob("stashed_*_properties_*.txt")), [])
+        mapping_file.unlink()
+        xml_file_replaced.unlink()
 
     def test_excel_to_json_project(self) -> None:
-        excel2json(
-            data_model_files="testdata/excel2json/excel2json_files",
-            path_to_output_file="testdata/tmp/_out_project.json",
+        excel_folder = Path("testdata/excel2json/excel2json_files")
+        out_file = Path("testdata/tmp/_out_project.json")
+        subprocess.run(
+            f"poetry run dsp-tools excel2json {excel_folder} {out_file.absolute()}",
+            check=True,
+            shell=True,
+            capture_output=True,
+            cwd=self.cwd,
         )
         with open("testdata/excel2json/excel2json-expected-output.json", encoding="utf-8") as f:
             output_expected = json.load(f)
-        with open("testdata/tmp/_out_project.json", encoding="utf-8") as f:
+        with open(out_file, encoding="utf-8") as f:
             output = json.load(f)
         self.assertDictEqual(output, output_expected)
-        os.remove("testdata/tmp/_out_project.json")
+        out_file.unlink()
 
     def test_excel_to_json_list(self) -> None:
```

Comment:
Starting from here, I substantially improve the tests: Instead of testing if the output file exists, I test its exact content.